terraform/modules/tf_s3_events/main.tf

```diff
@@ -26,7 +36,7 @@ resource "aws_s3_bucket_notification" "bucket_notification" {
 
 // IAM Policy: Allow Lambda to GetObjects from S3
 resource "aws_iam_role_policy" "lambda_s3_permission" {
-  name   = "${var.prefix}_${var.cluster}_S3GetObjects_${var.bucket_name}"
+  name   = "S3GetObjects_${var.bucket_name}"
```

Comment:
Why don't we also use `local.sanitized_bucket_name` instead of `var.bucket_name`?

Reply:
because the `name` attribute for these resources support special characters (eg - characters that are NOT in the set `[a-zA-Z0-9_-]`).. while the `statement_id` attribute about only supports those characters: https://docs.aws.amazon.com/lambda/latest/dg/API_AddPermission.html#SSS-AddPermission-request-StatementId

Reply:
Aha, this is annoy that we have to be very accurate to know which resource name needs to be sanitized 😒 Thanks for the explanation!